### DISCUSSION TITLE
Way to convert `pgx.Row` to `pgx.CollectableRow`? Or how to deal with unknown model schema?

### DISCUSSION THREAD

**Discussion by shabemdadi** [jackc/pgx#1906] [Q&A] [answered]
We have an insert command with a `RETURNING..` at the end. From looking here https://github.com/jackc/pgx/issues/411, it looks like we should be using the `QueryRow` functionality. 

However, unlike the `Query` command which returns `pgx.Rows` and which I can use the `RowToMap` functionality to do something like this where I do not need to know that underlying model schema.

```
results := make([]any, 0)
pgx.CollectRows(rows, func(row pgx.CollectableRow) (any, error) {
			thisMap, err := pgx.RowToMap(row)
			if err != nil {
				return results, nil
			}
			results = append(results, thisMap)
			return results, nil
		})
```

The `QueryRow` method returns `pgx.Row` which does not look like it has the same functionality and requires and explicit `Scan`

Is there a way to convert `pgx.Row` to `pgx.CollectableRow`? Or, another way around this?

**Comment by jackc** [comment, marked as answer]
Maybe use `Query` and `Collect(Exactly)OneRow`?

**Reply by shabemdadi** [reply]
oh I didn't realize you could do INSERT operations with `Query` - great I will give that a try then
